### PR TITLE
[config] support waypoint from storage

### PR DESCRIPTION
**Pull request by davidiw**
Per title, this is the last PR in this series before we make config
depend on secure storage and clean up this fun little mess.

### CODE REVIEW

File: secure/storage/src/config.rs [hunk 1]
```diff
@@ -46,3 +41,21 @@ pub fn peer_id(config: &NetworkConfig) -> PeerId {
...
+                .get(libra_global_constants::WAYPOINT)
+                .expect("Unable to read waypoint")
+                .value
+                .string()
+                .expect("Expected string for waypoint");
+            Some(Waypoint::from_str(&waypoint).expect("Unable to parse waypoint"))
+        }
+        WaypointConfig::None => None,
+    };
+    waypoint.expect("peer id should be present")
```
Comment (mgorven) [inline secure/storage/src/config.rs:60]: s/peer id/waypoint/ here?

Comment (davidiw) [reply secure/storage/src/config.rs:60]: copypasta :D
good call!

File: state-synchronizer/src/synchronizer.rs [hunk 2]
```diff
@@ -60,7 +61,7 @@ impl StateSynchronizer {
             network,
             state_sync_to_mempool_sender,
             config.base.role,
-            config.base.waypoint,
+            Some(waypoint),
```
Comment (zekun000) [inline state-synchronizer/src/synchronizer.rs:64]: thanks for fixing it, why not fix all the way?

Comment (davidiw) [reply state-synchronizer/src/synchronizer.rs:64]: not the right PR for that :D

**Pull request closed without merging**